powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/util/HashBasedCounterUtils.java

```diff
@@ -0,0 +1,110 @@
+/*
+ * PowerAuth Crypto Library
+ * Copyright 2019 Wultra s.r.o.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.getlime.security.powerauth.crypto.lib.util;
+
+import io.getlime.security.powerauth.crypto.lib.config.PowerAuthConfiguration;
+import io.getlime.security.powerauth.crypto.lib.generator.KeyGenerator;
+import io.getlime.security.powerauth.crypto.lib.model.exception.GenericCryptoException;
+import io.getlime.security.powerauth.provider.exception.CryptoProviderException;
+
+import javax.crypto.SecretKey;
+import java.nio.ByteBuffer;
+import java.security.InvalidKeyException;
+import java.util.Arrays;
+
+/**
+ * The {@code HashBasedCounterUtils} class provides additional functionality that allows secure transfer
+ * and verification of hash based counter over the network.
+ */
+public class HashBasedCounterUtils {
+
+    private final KeyGenerator keyGenerator;
+
+    /**
+     * @param keyGenerator {@link KeyGenerator} instance to be used in the class.
+     */
+    public HashBasedCounterUtils(KeyGenerator keyGenerator) {
```

Comment:
Do we really need to pass `KeyGenerator` as parameter? We can just use `keyGenerator = new KeyGenerator` in field initialization, there is no state required to be passed and creating new instance is cheap.

Reply:
Removed also from previously added `KeyDerivationUtils`